packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -499,12 +499,24 @@ class DevFS {
 
     // Update modified files
     final Map<Uri, DevFSContent> dirtyEntries = <Uri, DevFSContent>{};
-
     int syncedBytes = 0;
+    if (fullRestart) {
+      generator.reset();
+    }
+    // On a full restart, or on an initial compile for the attach based workflow,
+    // this will produce a full dill. Subsequent invocations will produce incremental
+    // dill files that depend on the invalidated files.
+    _logger.printTrace('Compiling dart to kernel with ${invalidatedFiles.length} updated files');
+    final Future<CompilerOutput> pendingCompilerOutput = generator.recompile(
```

Comment:
Maybe add a comment here explaining why the `Future` is awaited later.

Reply:
Done